src/main/java/com/google/sps/data/DeliverySystem.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.maps.errors.ApiException;
+import com.google.sps.data.Point;
+import java.io.IOException;
+import java.lang.InterruptedException;
+import java.util.Date;
+import java.util.HashMap;
+
+public class DeliverySystem {
+  /** Variable pathFinder will be used for getting the path between two Point objects.
+   * private PathFinder pathFinder;
+   *
+   * public DeliverySystem(PathFinder pathFinder) {
+   *   this.pathFinder = pathFinder;
+   * }
+   */
+
+  /**
+   * Assigns a set of orders for the delivery request such that they fit the [startTime, endTime]
+   * timeslot and the maximum number of stops the courier will make is <= maxWaypoints
+   */
+  public Journey findOrdersForDeliveryRequest(DeliverySlot deliverySlot, int maxWaypoints, Point start)  throws ApiException, IOException, InterruptedException {
+    throw new UnsupportedOperationException("TODO: Implement this method.");
+  }
+
+  /**
+   * Finds the order in which points must be visited such that the journey lasts minimum time and it
+   * delivers all orders assigned in findOrdersForDeliveryRequest
+   */
+  public Journey findOptimalDeliveryJourneyForRequest(DeliverySlot deliverySlot, int maxWaypoints, String userId, Double startLat, Double startLng) throws ApiException, IOException, InterruptedException {
```

Comment:
`findOrdersForDeliveryRequest` queries datastore to find orders in the courier's area. Then, it assigns at most 25 orders such that the optimal journey fits [startTime, endTime]. The order in which points should be visited is computed in `findOptimalDeliveryJourneyForRequest`. I used both methods so that I can test them independently: 

- `findOrdersForDeliveryRequest` only checks that for the assigned order, there is a journey that lasts at most endTime - startTime seconds
- `findOptimalDeliveryJourneyForRequest` finds the optimal order. It returned a Journey object because I thought that more details would be useful for displaying the journey.

If it makes more sense, I will keep `findOptimalDeliveryJourneyForRequest` and change it to return the points in optimal order.